airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py

```diff
@@ -126,3 +148,28 @@ def spec(self, *args, **kwargs) -> ConnectorSpecification:
                 ),
             ),
         )
+
+    def _update_insights_streams(self, insights, args, streams) -> List[Type[Stream]]:
+        """Update method, if insights have values returns streams replacing the
+        default insights streams else returns streams
+
+        """
+        if not insights:
+            return streams
+
+        insights_custom_streams = list()
+
+        for insight in insights:
+            args["name"] = insight.name
```

Comment:
is this mutating the same config over and over, so in the end all the streams have the same config? can we copy the args instead? e.g: 

```
arg_copy = copy.deepcopy(args)
arg_copy["fields"] = ...
```